frontend/src/lib/components/Input/input.tsx

```diff
@@ -106,6 +106,22 @@ export const Input = React.forwardRef((props: InputProps, ref: React.ForwardedRe
         setValue(event.target.value);
 
         if (props.onChange) {
+            if (props.type === "number") {
+                let newValue = 0;
+
+                if (!isNaN(parseFloat(event.target.value as string))) {
+                    newValue = parseFloat((event.target.value as string) || "0");
+                    if (props.min !== undefined) {
+                        newValue = Math.max(props.min, newValue);
+                    }
+
+                    if (props.max !== undefined) {
+                        newValue = Math.min(props.max, newValue);
+                    }
+                }
+
+                event.target.value = newValue.toString();
+            }
```

Comment:
Should the `setValue` on line 106 be triggered after adjustment/configuration of `event.target.value`, i.e. move the if-check and value adjustment outside of the `props.onChange`-if and before the `setValue`?

Reply:
As discussed yesterday, they should not be synced in order to allow intermediate invalid states locally without publishing them to parent.